src/main.rs

```diff
@@ -0,0 +1,289 @@
+    #[serde(rename = "maxPoints")]
+    max_points: usize,
+}
+
+/// Response returned to report error conditions
+#[derive(Serialize, Debug)]
+struct ErrorResponse {
+    /// Human-readable description of the error
+    message: String,
+}
+
+/// Server error conditions
+#[derive(Debug)]
+enum Error {
+    LockFailure,
+    BadPoint,
+    TooManyPoints,
+    BadEpoch(u8),
+    Base64(base64::DecodeError),
+    Oprf(ppoprf::PPRFError),
+}
+
+impl axum::response::IntoResponse for Error {
+    fn into_response(self) -> axum::response::Response {
+        let message = match self {
+            Error::LockFailure =>
+                "Couldn't lock state: RwLock poisoned".into(),
+            Error::BadPoint =>
+                "Invalid point".into(),
+            Error::TooManyPoints =>
+                "Too many points for a single request".into(),
+            Error::BadEpoch(epoch) =>
+                format!("Invalid epoch {epoch}"),
+            Error::Base64(e) =>
+                format!("invalid base64 encoding: {e}"),
+            Error::Oprf(e) =>
+                format!("PPOPRF error: {e}"),
+        };
+        let body = Json(ErrorResponse { message });
+        (StatusCode::BAD_REQUEST, body).into_response()
+    }
+}
+
+/// Process PPOPRF evaluation requests
+async fn randomness(
+    State(state): State<OPRFState>,
+    Json(request): Json<RandomnessRequest>,
+) -> Result<Json<RandomnessResponse>, Error> {
+    debug!("recv: {request:?}");
+    let state = state.read().map_err(|_| Error::LockFailure)?;
+    let epoch = request.epoch.unwrap_or(state.epoch);
+    if epoch != state.epoch {
+        return Err(Error::BadEpoch(epoch));
+    }
+    if request.points.len() > MAX_POINTS {
+        return Err(Error::TooManyPoints);
+    }
+    let prove = false;
+    let mut points = Vec::with_capacity(request.points.len());
+    for base64_point in request.points {
+        let input = BASE64.decode(base64_point)
+            .map_err(Error::Base64)?;
+        // FIXME: Point::from is fallible and needs to return a result.
+        // partial work-around: check correct length
+        if input.len() != ppoprf::COMPRESSED_POINT_LEN {
+            return Err(Error::BadPoint);
+        }
+        let point = ppoprf::Point::from(input.as_slice());
+        let evaluation = state.server.eval(&point, epoch, prove)
+            .map_err(Error::Oprf)?;
+        points.push(BASE64.encode(evaluation.output.as_bytes()));
+    }
+    let response = RandomnessResponse { points, epoch };
+    debug!("send: {response:?}");
+    Ok(Json(response))
+}
+
+/// Process PPOPRF epoch and key requests
+async fn info(
+    State(state): State<OPRFState>
+) -> Result<Json<InfoResponse>, Error> {
+    debug!("recv: info reqeust");
+    let state = state.read().map_err(|_| Error::LockFailure)?;
+    let current_epoch = state.epoch;
+    // FIXME: return the end of the current epoch
+    let next_epoch_time = "unknown".to_owned();
+    let max_points = MAX_POINTS;
```

Comment:
Saves writing `max_points: MAX_POINTS` initializing the response struct.

Reply:
totally a nit, but we would save one line by removing it? could make things easier to read since one does not have to infer the purpose of the variable when reviewing